rust_src/src/lisp.rs

```diff
@@ -130,13 +141,31 @@ pub enum LispType {
     Lisp_Float = 7,
 }
 
+impl LispObject {
+    #[allow(unused_unsafe)]
+    pub unsafe fn get_type_unchecked(self) -> LispType {
```

Comment:
Should this be an unsafe function? I believe it's safe: you can't pass it a value that will break memory safety.